### ISSUE TITLE
Refactoring of Expedient plugin system

### ISSUE THREAD

**Issue by msune** [fp7-ofelia/ocf#68]
Refactoring of Expedient plugin system to have:

1) Well defined workflow (HTML) per plugin
2) Homogenous and easy to use API to use the Topology map in the visualization panel
3) Simple and easy to understand relation with XMLRPCs clients and RSPECs and similar

**Comment by CarolinaFernandez** [comment]
## Notes for development

As for the resource visualization (see slice detail), there will be a common "topology map engine" that will act as the **core plugin** for every resource visualization: the resources plugins (e.g. OpenFlow switch, PlanetLab switch, Server, Virtual Machine...) shall implement a **common interface** from which the topology engine will **perform standard calls to ask for these resources** (e.g. the links -- port X is connected to resource with datapathID Y).

```
Slice: [<Resource #1, Type OpenflowSwitch>, <Resource #2, Type Server>]

Topology map engine: getLinks (Resource #1) -----> OpenflowSwitch plugin: (specific logic to retrieve links; return links (Resource #1)) -----> Topology map engine: drawLinks (Resource #1)

Topology map engine: getLinks (Resource #2) -----> Server plugin: (specific logic to retrieve links; return links (Resource #2)) -----> Topology map engine: drawLinks (Resource #2)
```

If some plugin needs to show the resources in its workflow (e.g. "Book Openflow resources" in Openflow switches), **then it would the plugin that should contact the Topology map engine** (and the corresponding files would be inside the plugin folder)

## Key idea

Avoid the need of new (specific) code for every new type of resource added to a slice ( _that should be transparent_ ): abstraction of Expedient plugin intercommunication.

(PS: Correct me if am wrong)

**Comment by CarolinaFernandez** [comment]
Done -- though it will be released along with v0.5.

Explanation in the wiki: [Developing plugins for Expedient](https://github.com/fp7-ofelia/ocf/wiki/Developing%20plugins%20for%20Expedient)
